pkg/ocm/ocm_config.go

```diff
@@ -0,0 +1,136 @@
+package ocm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+)
+
+// Config is the type used to store the configuration of the client.
+// There's no way to line-split or predefine tags, so...
+//nolint:lll
+type Config struct {
+	// TODO(efried): Better docs for things like AccessToken
+	// TODO(efried): Dedup with flag docs in cmd/ocm/login/cmd.go:init where possible
+	AccessToken  string   `json:"access_token,omitempty" doc:"Bearer access token."`
+	ClientID     string   `json:"client_id,omitempty" doc:"OpenID client identifier."`
+	ClientSecret string   `json:"client_secret,omitempty" doc:"OpenID client secret."`
+	Insecure     bool     `json:"insecure,omitempty" doc:"Enables insecure communication with the server. This disables verification of TLS certificates and host names."`
+	Password     string   `json:"password,omitempty" doc:"User password."`
+	RefreshToken string   `json:"refresh_token,omitempty" doc:"Offline or refresh token."`
+	Scopes       []string `json:"scopes,omitempty" doc:"OpenID scope. If this option is used it will replace completely the default scopes. Can be repeated multiple times to specify multiple scopes."`
+	TokenURL     string   `json:"token_url,omitempty" doc:"OpenID token URL."`
+	URL          string   `json:"url,omitempty" doc:"URL of the API gateway. The value can be the complete URL or an alias. The valid aliases are 'production', 'staging' and 'integration'."`
+	User         string   `json:"user,omitempty" doc:"User name."`
+	Pager        string   `json:"pager,omitempty" doc:"Pager command, for example 'less'. If empty no pager will be used."`
+}
+
+// Load loads the configuration from the configuration file. If the configuration file doesn't exist
+// it will return an empty configuration object.
+func Load() (cfg *Config, err error) {
```

Comment:
Can we not use the Load function from ocm-cli and remove this part, as well as the the two function below? Sothat we at least are in sync with what ocm-cli expects and everyone uses as a config? I mean the PR looks way bigger than what was removed earlier in: https://github.com/openshift/configuration-anomaly-detection/commit/52bd34d9b67ad8c418d3c069b935dfe9d782d571

Reply:
Yes, this is intended. @rafael-azevedo said that he does not want to rely on ocm-cli, because ocm-cli was never intended for library usage. This is why I pulled in the changes manually, this way we are in the position to modify it and maintain it.

We can just use the ocm-cli library, but then we will pull in the nil pointer problem from ocm-cli + we will adopt any chanegs ocm-cli will do. The ocm-cli devs were pretty clear about this that we should not use their code as library